dewy/document/router.py

```diff
@@ -87,11 +87,11 @@ async def add_document(
         row = None
         row = await conn.fetchrow(
             """
-        INSERT INTO document (collection_id, url, ingest_state)
-        VALUES ($1, $2, 'pending')
-        RETURNING id, collection_id, url, ingest_state, ingest_error
-        """,
-            req.collection_id,
+            INSERT INTO document (collection_id, url, ingest_state)
+            VALUES ((SELECT id FROM collection WHERE name = $1), $2, 'pending')
+            RETURNING id, collection_id, url, ingest_state, ingest_error, $1 AS collection
+            """,
+            req.collection,
```

Comment:
This is using the collection name as the ID

Reply:
No. It is passing the collection name to `(SELECT id FROM collection WHERE name = $1)` and uses the resulting row as the collection ID.

Reply:
Added tests verifying this was working, although it did need to be `lower(name) = lower($1)` to ensure our case insensitivity.

Reply:
Also, this would have been verified by the existing tests (eg., we couldn't add a document if it didn't work).